src/App.tsx

```
@@ -1,26 +1,248 @@
+
+  const returnLeftNumber = () => {
+    return todos.filter(todo => todo.completed != true).length;
+  };
+
+  const handleSetQuery = (passedQuery: FilterType) => {
+    setQuery(passedQuery);
+  };
+
+  const handleSetError = (errorType: ErrorType | null) => {
+    setError(errorType);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (inputRef.current?.value.trim()) {
+      inputRef.current.disabled = true;
+      setTmpTodo({
+        title: inputRef.current.value.trim(),
+        id: -1,
+        completed: false,
+      });
+
+      postTodo(inputRef.current.value.trim())
+        .then(response => {
+          setTodos(prev => [...prev, response]);
+
+          if (inputRef.current) {
+            inputRef.current.value = '';
+          }
+        })
+        .catch(() => setError(ErrorType.UnableToAdd))
+        .finally(() => {
+          if (inputRef.current) {
+            inputRef.current.disabled = false;
+            inputRef.current.focus();
+          }
+
+          setTmpTodo(null);
+        });
+    } else {
+      setError(ErrorType.EmptyTitle);
+    }
+  };
+
+  const handleRemoveTodoById = async (id: number): Promise<boolean> => {
+    setUpdatedTodosId(prev => [...prev, id]);
+    let success = false;
+
+    try {
+      await removeTodo(id);
+
+      setTodos(prev => prev.filter(el => el.id != id));
+      inputRef.current?.focus();
+      success = true;
+    } catch {
+      handleSetError(ErrorType.UnableToDelete);
+    } finally {
+      setUpdatedTodosId(prev => prev.filter(el => el != id));
+    }
+
+    /* removeTodo(id)
+      .then(() => {
+        setTodos(prev => prev.filter(el => el.id != id));
+        success = true;
+      })
+      .catch(() => setError(ErrorType.UnableToDelete))
+      .finally(() => {
+        setUpdatedTodosId(prev => prev.filter(el => el != id));
+        inputRef.current?.focus();
+      });
+    */
+    return success;
```

do you realy need this variable and return, think about how to avoid this

---

I do not see better solution, only other is flag which had be te cleaned each time and may interact with other element

---

I'm using here return success to not close input during editing title while leaving it empty

---

You can just rethrow the error inside catch block and catch in another place.